src/interfaces/IOracles.sol

```diff
@@ -0,0 +1,147 @@
+     */
+    function report(address rateFeedId) external;
+
+    /**
+     * @notice Sets `hasFreshness` to `false` if the most recent report has
+     * become outdated.
+     * @param rateFeedId The rate feed to mark stale.
+     */
+    function markStale(address rateFeedId) external;
+
+    /**
+     * @notice Sets the window size over which a rate feed's medians will be
+     * averaged.
+     * @param rateFeedId The rate feed being configured.
+     * @param windowSize The number of most recent medians to average over for
+     * the final reported median.
+     * @dev Only callable by the owner.
+     */
+    function setWindowSize(address rateFeedId, uint8 windowSize) external;
+    /**
+     * @notice Sets the allowed deviation for a rate feed.
+     * @param rateFeedId The rate feed being configured.
+     * @param allowedDeviation The maximal multiplicative deviation allowed
+     * between two values in a report batch, expressed as the numerator of a
+     * fraction over uint16.max.
+     * @dev Only callable by the owner.
+     */
+    function setAllowedDeviation(
+        address rateFeedId,
+        uint16 allowedDeviation
+    ) external;
+    /**
+     * @notice Sets the required quorum for a rate feed.
+     * @param rateFeedId The rate feed being configured.
+     * @param quorum The minimum number of individual reporters that need to be
+     * present in a report batch.
+     * @dev Only callable by the owner.
+     */
+    function setQuorum(address rateFeedId, uint8 quorum) external;
+    /**
+     * @notice Sets the certainty threshold for a rate feed.
+     * @param rateFeedId The rate feed being configured.
+     * @param certaintyThreshold The minimum number of values that need to be
+     * doneted as certain in a batch for it to be considered valid.
+     * @dev Only callable by the owner.
+     */
+    function setCertaintyThreshold(
+        address rateFeedId,
+        uint8 certaintyThreshold
+    ) external;
+    /**
+     * @notice Sets the allowed staleness for a rate feed.
+     * @param rateFeedId The rate feed being configured.
+     * @param allowedStaleness The number of seconds before a report becomes
+     * considered stale and no longer valid.
+     * @dev Only callable by the owner.
+     */
+    function setAllowedStaleness(
+        address rateFeedId,
+        uint16 allowedStaleness
+    ) external;
+
+    /**
+     * @notice Adds a new supported rate feed.
+     * @param rateFeedId The new rate feed's ID.
+     * @dev Only callable by the owner.
+     */
+    function addRateFeed(address rateFeedId) external;
+    /**
+     * @notice Removes a rate feed.
+     * @param rateFeedId The rate feed's ID.
+     * @dev Only callable by the owner.
+     */
+    function removeRateFeed(address rateFeedId) external;
+
+    /**
+     * @notice Adds a new trusted provider, i.e. a new offchain oracle node
```

Comment:
I'm undecided if we should change the terminology here. I see that Redstone sometimes uses 'provider' and other times 'oracle node operator'. And `SortedOracles` just calls it 'oracle'.

Personally, just 'provider' sounds a bit too fuzzy to my ears. Wdyt about these?

- `addOracle`
- `addOracleOperator`
- `addDataProvider`